test/utils/browser.js

```
@@ -30,32 +42,83 @@ exports.openMusic = function (options) {
+      that.browser.maximize();
+      that.browser.setAsyncScriptTimeout(ASYNC_SCRIPT_TIMEOUT, done);
+    });
   });
   before(function navigateToMusicBeforeLogin (done) {
     this.browser.get(url, done);
   });
-  before(function handleLoginViaCookies (done) {
+  before(function handleLogin (done) {
     var browser = this.browser;
-    async.forEach(cookies, function setCookies (cookie, cb) {
-      // If the cookie is not for .google.com, skip it
-      // DEV: As discovered by Burp suite's repeater, we only need `SID`, `HSID`, `SSID` but this is simpler
-      if (cookie.domain !== '.google.com') {
-        process.nextTick(cb);
-      // Otherwise, set it
-      } else {
-        browser.setCookie(cookie, cb);
+    var that = this;
+
+    async.waterfall([
+      function findSignInButton (cb) {
+        browser.waitForElementByCssSelector('[data-action=signin]', asserters.isDisplayed, cb);
+      },
+      function clickSignInButton (el, cb) {
+        el.click(cb);
+      },
+      function findEmailInput (cb) {
+        browser.waitForElementById('Email', asserters.isDisplayed, cb);
+      },
+      function enterEmailIntoInput (el, cb) {
+        exports.simpleExecute.call(that, function setEmailInputValue () {
+          document.getElementById('Email').value = arguments[0];
+        }, [GOOGLE_MUSIC_JS_EMAIL], cb);
+      },
+      function findNextButton (err, cb) {
+        browser.waitForElementById('next', asserters.isDisplayed, cb);
+      },
+      function clickNextButton (el, cb) {
+        el.click(cb);
+      },
+      function findPasswordInput (cb) {
+        browser.waitForElementById('Passwd', asserters.isDisplayed, cb);
+      },
+      function enterPasswordIntoInput (el, cb) {
+        exports.simpleExecute.call(that, function setPasswordInputValue () {
+          document.getElementById('Passwd').value = arguments[0];
+        }, [GOOGLE_MUSIC_JS_PASSWORD], cb);
+      },
+      function findLoginButton (err, cb) {
```

Why are we accepting an error here? That should be getting swallowed by `async.waterfall` =_=